src/storage/src/mem_table.rs

```diff
@@ -81,18 +85,25 @@ impl MemTable {
     /// write methods
     pub fn insert(&mut self, pk: Bytes, value: Bytes) -> Result<()> {
         if !self.is_consistent_op {
-            self.buffer.insert(pk, KeyOp::Insert(value));
+            let key_len = pk.len();
+            self.kv_size.add(&pk, &value);
+            let origin_value = self.buffer.insert(pk, KeyOp::Insert(value));
+            self.sub_origin_size(origin_value, key_len);
+
             return Ok(());
         }
         let entry = self.buffer.entry(pk);
         match entry {
             Entry::Vacant(e) => {
+                self.kv_size.add(e.key(), &value);
                 e.insert(KeyOp::Insert(value));
                 Ok(())
             }
             Entry::Occupied(mut e) => match e.get_mut() {
                 KeyOp::Delete(ref mut old_value) => {
                     let old_val = std::mem::take(old_value);
+
+                    self.kv_size.add_val(&value);
```

Comment:
Did we miss to subtract the old value size here?

Reply:
The old value is stored in `Delete(old_value)`, and here mem_table just convert Delete to Update, so we only need to add new value.